sopel/modules/announce.py

```diff
@@ -9,14 +9,24 @@
 """
 from __future__ import unicode_literals, absolute_import, print_function, division
 
+from builtins import range
+
 from sopel.module import commands, example, require_admin
 
 
+# from https://stackoverflow.com/a/312464/5991 with modified names for readability
+def chunks(iterbl, size):
```

Comment:
There is something important that is missing here: `iterbl` **must have** a length, which isn't the case for a generator.

In that regard, I suggest two actions:

* rename from "iterbl" (which I assume is a shortcut (which is not good to me) for "iterable") to "items"
* add a docstring that explains this

I know, 2 lines of code, *so many* consequences!

Reply:
Seconde note: this could be in tools/utils.

Reply:
The original was built to work with lists, and named the argument `lst`. This is kind of trivial to put in `tools`, IMO. It's just meant to be a helper (and I considered simplifying this to a comprehension expr, but made it separate for eventual unit testing).

I don't really care about the parameter name(s), but I object to excessively bulletproofing a one-off helper that nobody should use because it's not part of a documented API… I'd sooner specialize it to take `bot` as an argument instead, and rename the function `get_channel_groups` or something.

Reply:
Oh, I think we can come with a good solution here: make it a bot's method instead! After all, the bot knows about targmax, it knows about channels, and it can do everything asked!

Reply:
What are you proposing as the API feature here, exactly? `bot.group_max_channels(*list)`?

OK, so naming things is hard. But we probably want to give serious thought to actually _designing_ API features around the basic ISUPPORT/MYINFO data Sopel tracks, rather than tossing them in at random based on trivial use cases like this.

Reply:
Yeah it is.

As "trivial" as it sound, this is exactly the right use-case for TARGMAX: take a list of targets, and yields each group of targets according to TARGMAX!

I'd do a method with 2 arguments:

```
def lets_find_a_name(self, *targets, max_target=None):
    """Yield group of ``targets`` according to ISUPPORT's TARGMAX or up to ``max_target``.

    <more details>
    """
    # 1. targmax = (TARGMAX or 1) if max_target is None else min(TARGMAX, max_target)
    # 2. chunk according to this targmax
```

No name yet, but I think you get the general idea. I think it's a good interface.

Reply:
Let's do this separately, and put some real discussion into it. I agree that it would be useful for the `bot` object or `tools` module to have something for this, but implementing this for `announce` is so trivial that it shouldn't be held up by a larger discussion about new API behavior/features.

Reply:
I now realize that this syntax: `(self, *targets, max_target=None)` ; is not supported in Py2. This will have to wait Sopel 8.